faros_event.sh

```diff
@@ -402,432 +345,280 @@ function resolveDefaults() {
-        '{
-            "compute_Application": {
-                "name": $app,
-                "platform": $deploy_app_platform
-            }
-        }'
-    )
+function parseDeployUri() {
+    parseUri "${deploy_uri:-$FAROS_DEPLOY}" "deploy_source" "deploy_app" "deploy_env" "deploy_id" $deploy_uri_form
 }
 
-function makeOrganization() {
-    cicd_Organization=$( jq -n \
-        --arg cicd_org "$cicd_org" \
-        --arg cicd_source "$cicd_source" \
-        '{
-            "cicd_Organization": {
-                "uid": $cicd_org,
-                "source": $cicd_source
-            }
-        }'
-    )
+function parseArtifactUri() {
+    parseUri "${artifact_uri:-$FAROS_ARTIFACT}" "artifact_source" "artifact_org" "artifact_repo" "artifact_id" $artifact_uri_form    
 }
 
 function makeEvent() {
     request_body=$( jq -n \
         --arg origin "$origin" \
+        --arg event_type "$event_type" \
         '{ 
+            "type": $event_type,
+            "version": "0.0.1",
             "origin": $origin,
-            "entries": []
         }'
     )
 }
 
-function addBuildToEvent() {
-    makeBuild
-    request_body=$(jq ".entries += [
-        $cicd_Build
-    ]" <<< $request_body)
+function addDeployToData() {
+    if ! [ -z "$deploy_source" ] &&
+       ! [ -z "$deploy_app" ] &&
+       ! [ -z "$deploy_env" ] &&
+       ! [ -z "$deploy_id" ]; then
+        request_body=$(jq \
+            --arg deploy_source "$deploy_source" \
+            --arg deploy_app "$deploy_app" \
+            --arg deploy_env "$deploy_env" \
+            --arg deploy_id "$deploy_id" \
+            '.data.deploy +=
+            {
+                "source": $deploy_source,
+                "application": $deploy_app,
+                "environment": $deploy_env,
+                "id": $deploy_id
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_status" ]; then
+        request_body=$(jq \
+            --arg deploy_status "$deploy_status" \
+            '.data.deploy +=
+            {
+                "status": $deploy_status
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_status_details" ]; then
+        request_body=$(jq \
+            --arg deploy_status_details "$deploy_status_details" \
+            '.data.deploy +=
+            {
+                "statusDetails": $deploy_status_details
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_env_details" ]; then
+        request_body=$(jq \
+            --arg deploy_env_details "$deploy_env_details" \
+            '.data.deploy +=
+            {
+                "environmentDetails": $deploy_env_details
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_start_time" ]; then
+        request_body=$(jq \
+            --arg deploy_start_time "$deploy_start_time" \
+            '.data.deploy +=
+            {
+                startTime: $deploy_start_time|tonumber
```

Comment:
`"startTime"`

Reply:
weird that this works like this. Will change it